shell/common/engine.h

```diff
@@ -767,6 +781,28 @@ class Engine final : public RuntimeDelegate,
   ///
   const std::string& InitialRoute() const { return initial_route_; }
 
+  //--------------------------------------------------------------------------
+  /// @brief      Loads the dart shared library from disk and into the dart VM
+  ///             based off of the search parameters. When the dart library is
+  ///             loaded successfully, the dart future returned by the
+  ///             originating loadLibrary() call completes.
+  ///
+  /// @param[in]  loading_unit_id  The unique id of the deferred library's
+  ///                              loading unit.
+  ///
+  /// @param[in]  lib_name         The file name of the .so shared library
+  ///                              file.
+  ///
+  /// @param[in]  apkPaths         The paths of the APKs that may or may not
+  ///                              contain the lib_name file.
+  ///
+  /// @param[in]  abi              The abi of the library, eg, arm64-v8a
+  ///
+  void CompleteDartLoadLibrary(intptr_t loading_unit_id,
+                               std::string lib_name,
+                               std::vector<std::string>& apkPaths,
```

Comment:
My initial plans were to keep the dlopen-ing where the other instances of dlopening libs are, but now that you mention it, it probably makes more sense to handle it in the android-engine and pass the handle along instead to avoid android specific stuff in the general engine